staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/streaming.go

```diff
@@ -0,0 +1,242 @@
+	}
+	listMetaValue := v.FieldByName("ListMeta")
+	if !listMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field")
+	}
+	listMeta, ok := listMetaValue.Interface().(metav1.ListMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected ListMeta field to have ListMeta type")
+	}
+	typeMetaField, ok := listType.FieldByName("TypeMeta")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	if typeMetaField.Tag.Get("json") != ",inline" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	typeMetaValue := v.FieldByName("TypeMeta")
+	if !typeMetaValue.IsValid() {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field")
+	}
+	typeMeta, ok := typeMetaValue.Interface().(metav1.TypeMeta)
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field to have TypeMeta type")
+	}
+
+	itemsField, ok := listType.FieldByName("Items")
+	if !ok {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected Items field")
+	}
+	if itemsField.Tag.Get("json") != "items" {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, fmt.Errorf("expected TypeMeta field tag")
+	}
+	items, err := meta.ExtractList(list)
+	if err != nil {
+		return metav1.TypeMeta{}, metav1.ListMeta{}, nil, err
+	}
+	return typeMeta, listMeta, items, nil
+}
+
+func streamingEncodeList(w io.Writer, typeMeta metav1.TypeMeta, listMeta metav1.ListMeta, items []runtime.Object) error {
+	// Start
+	if _, err := w.Write([]byte(`{`)); err != nil {
+		return err
+	}
+
+	// TypeMeta
+	if typeMeta.Kind != "" {
+		if err := encodeKeyValuePair(w, "kind", typeMeta.Kind, true); err != nil {
+			return err
+		}
+	}
+	if typeMeta.APIVersion != "" {
+		if err := encodeKeyValuePair(w, "apiVersion", typeMeta.APIVersion, true); err != nil {
+			return err
+		}
+	}
+
+	// ListMeta
+	if err := encodeKeyValuePair(w, "metadata", listMeta, true); err != nil {
+		return err
+	}
+
+	// Items
+	if err := encodeItemsObjectSlice(w, items, false); err != nil {
+		return err
+	}
+
+	// End
+	_, err := w.Write([]byte("}\n"))
+	return err
+}
+
+func encodeItemsObjectSlice(w io.Writer, items []runtime.Object, sep bool) (err error) {
+	if items == nil {
+		err := encodeKeyValuePair(w, "items", nil, sep)
+		return err
+	}
+	_, err = w.Write([]byte(`"items":[`))
+	if err != nil {
+		return err
+	}
+	for i, item := range items {
+		err := encodeValue(w, item, i != len(items)-1)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = w.Write([]byte("]"))
+	if err != nil {
+		return err
+	}
+	if sep {
+		_, err = w.Write([]byte(","))
+	}
+	return err
+}
+
+func streamingEncodeUnstructuredList(w io.Writer, list *unstructured.UnstructuredList) error {
+	_, err := w.Write([]byte(`{`))
+	if err != nil {
+		return err
+	}
+	keys := slices.Sorted(maps.Keys(list.Object))
```

Comment:
adding `items` to the set of extracted keys if not already present would ensure we always had an explicit `items` entry to sort / handle inside the loop and would make the logic a lot easier to follow inside the loop

```go
// Get existing keys
keys := maps.Keys(list.Object)
// Ensure an "items" key is always included
if _, exists := list.Object["items"]; !exists {
  keys = append(keys, "items")
}
// Sort keys to match standard unstructured encoding
keys = slices.Sorted(keys)
```

Reply:
Done